src/libraries/System.Net.Http/src/System/Net/Http/Headers/MediaTypeHeaderValue.cs

```diff
@@ -47,14 +50,18 @@ public class MediaTypeHeaderValue : ICloneable
                     }
                     else
                     {
-                        Parameters.Add(new NameValueHeaderValue(charSet, value));
+                        Parameters.Add(new NameValueHeaderValue(CharSetName, value));
                     }
                 }
             }
         }
 
+        /// <summary>Gets or sets the media-type header value parameters.</summary>
```

Comment:
The property seems to be get-only
```suggestion
        /// <summary>Gets the media-type header value parameters.</summary>
```